bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingFactory.java

```diff
@@ -105,9 +108,13 @@ public static Thing createThing(ThingUID thingUID, Configuration configuration,
         for (ThingHandlerFactory thingHandlerFactory : thingHandlerFactories) {
             if (thingHandlerFactory.supportsThingType(thingTypeUID)) {
                 Thing thing = thingHandlerFactory.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
-                if (properties != null) {
-                    for (String key : properties.keySet()) {
-                        thing.setProperty(key, properties.get(key));
+                if (thing == null) {
+                    logger.error("Thing factory returned null when it reports to support the thing");
```

Comment:
Okay, I do not know of all the messages above and below this section.
In production systems I set the log level to error or warning (I think this is the common case), so I will not see any debug, info, ... messages.
This error should never be thrown in the wild, so I see no problem to log here a lot of things (because the branch should never be executed).

If the information is already logged, all fine.
But analyse some logs some times a user (with multiple bindings) provides that contains only this message if nothing that helps to identify the binding etc.